Modules/itertoolsmodule.c

```diff
@@ -186,14 +186,20 @@ static PyObject *
 batched_next(batchedobject *bo)
 {
     Py_ssize_t i;
-    Py_ssize_t n = bo->batch_size;
+    Py_ssize_t n = FT_ATOMIC_LOAD_SSIZE_RELAXED(bo->batch_size);
     PyObject *it = bo->it;
     PyObject *item;
     PyObject *result;
 
+#ifdef Py_GIL_DISABLED
     if (it == NULL) {
         return NULL;
     }
+#else
+    if (n < 0) {
```

Comment:
you can rely on this in both builds, the clearing of iterable is an optimization just in case of gil builds